File: src/routers/models/jobs.py
from typing import List, Optional

from pydantic import BaseModel, Field


class JobCategory(BaseModel):
    slug: str
    description: str
    parent__slug: Optional[str]
    parent__description: Optional[str]


class JobData(BaseModel):
    public_id: str
    description: str


class JobResponseData(BaseModel):
    public_id: str
    category: str


class JobCategorizationRequest(BaseModel):
    data: List[JobData] = Field(default=[], description="The data to categorize.")


class JobCategorizationResponse(BaseModel):
    data: List[JobResponseData] = Field(default=[], description="The categorized data.")
    error_message: Optional[str] = Field(default=None, description="The error message.")
    success: bool = Field(
        default=True, description="Whether the request was successful."
    )

File: internal/models/jobs.go
package models

// JobCategory represents a job category from the backend
type JobCategory struct {
	Slug              string  `json:"slug"`
	Description       string  `json:"description"`
	ParentSlug        *string `json:"parent_slug,omitempty"`
	ParentDescription *string `json:"parent_description,omitempty"`
}

// JobData represents input job data for categorization
type JobData struct {
	PublicID    string `json:"public_id" binding:"required"`
	Description string `json:"description" binding:"required"`
}

// JobResponseData represents categorized job output
type JobResponseData struct {
	PublicID string `json:"public_id"`
	Category string `json:"category"`
}

// JobCategorizationRequest is the request payload for job categorization
type JobCategorizationRequest struct {
	Data []JobData `json:"data" binding:"required,dive"`
}

// JobCategorizationResponse is the response for job categorization
type JobCategorizationResponse struct {
	Data         []JobResponseData `json:"data"`
	ErrorMessage *string           `json:"error_message,omitempty"`
	Success      bool              `json:"success"`
}
